### ISSUE TITLE
Not reading `$HOME/.exrc`

### ISSUE THREAD

**Issue by CRThaze** [n-t-roff/heirloom-ex-vi#10]
Version: `master` @ `bd1f886598d20d457885c5c189fea34b027cbdef`

It seems that I can't use `$HOME/.exrc` to configure the editor. No settings I provide seem to have any effect.

`$HOME/.exrc`:
```vim
set ts=2
set sw=2
set number
set showmode
set showmatch
set autoindent
```
Result:
![image](https://github.com/user-attachments/assets/864f71dc-4fe8-48d9-b345-759e3b1cb226)

The `EXINIT` environment variable does happen to work. So I was able to create the following workaround:

```bash
inline_exrc ()
{
    local file_path="$1"
    local processed_content=$(sed '/^$/d; /^"/d' "$file_path" | tr '\n' '|')
    processed_content=$(echo "$processed_content" | sed 's/|$//')
    echo "$processed_content"
}

alias vi='EXINIT="$(inline_exrc $HOME/.exrc)" vi'
```

Workaround result:
![image](https://github.com/user-attachments/assets/64d656d5-de42-4c8c-9ce1-649c86460f89)

**Comment by seal331** [comment]
I've done some debugging on this issue. It seems like this is intentional behavior - in order to check whether to source the file or not, it's calling a function called iownit, which returns 0 (aka do not read) if the condition ``(sb.st_mode & (S_IWOTH | S_IWGRP)`` is satisfied. This condition explicitly checks for the permissions of the .exrc file to have S_IWOTH, S_IWGRP or both to be set - and if any of those are true, the file's not being sourced. S_IWOTH is the bit that specifies the write permission for other users, and S_IWGRP is the bit that specifies the write permission for the group owner.

TLDR: chmod your .exrc to 644

P. S. there are also a couple other conditions being checked by the function in the case of "do not read", the only other one you really need to know about is ``(sb.st_uid != getuid())``, which checks if the owner of the file is the current user. If you want the file to be read under any permissions, you have to recompile with the SOURCEANY option set.
